pkg/virt-launcher/virtwrap/nichotplug.go

```diff
@@ -194,10 +199,11 @@ func withNetworkIfacesResources(vmi *v1.VirtualMachineInstance, domainSpec *api.
 }
 
 func appendPlaceholderInterfacesToTheDomain(vmi *v1.VirtualMachineInstance, domainSpec *api.DomainSpec) *api.DomainSpec {
+	if len(vmi.Spec.Domain.Devices.Interfaces) == 0 {
+		return domainSpec
+	}
 	domainSpecWithIfacesResource := domainSpec.DeepCopy()
-	requests := resource.ExtendedResourceList{ResourceList: vmi.Spec.Domain.Resources.Requests}
-	reqInterfaces := int(requests.Interface().Value())
-	interfacePlaceholderCount := reqInterfaces - len(vmi.Spec.Domain.Devices.Interfaces)
+	interfacePlaceholderCount := ReservedInterfaces - len(vmi.Spec.Domain.Devices.Interfaces)
```

Comment:
Just to understand, given a VMI with 5 interfaces, is it possible to hotplug another interface?

Reply:
No extra reservation is provided, only the default one that comes with q35 machine type.

We could set it to reserve 4 beyond the initially defined ones, but it is not clear if it is needed or not.
Not only that a user needs hotplug, but also starts with multiple interface.

Anyway, this can be changed per need. Even the current reserved number is guessed.
We should monitor the user needs and adjust accordingly.